File: Strings/Palindromic_Substring/PalindromicSubstring.java
import java.util.*;

public class PalindromicSubstring {

    public static void main(String... args) {
        System.out.println(palindromicSubstring("hellosannasmith").equals("sannas"));
        System.out.println(palindromicSubstring("abcdefgg").equals("none"));
    }

        public static boolean isPalindrome(String s) {
        for(int i = 0, j = s.length() - 1; i <= j; i++, j--) {
            if(s.charAt(i) != s.charAt(j)) {
                return false;
            }
        }
        return true;
    }

    public static String palindromicSubstring(String str) {
        SortedMap<Integer, String> palindromicSubstringCounter = new TreeMap<>();
        for(int i = 0; i < str.length(); i++) {
            char c1 = str.charAt(i);
            for(int j = str.length() - 1; j >= i; j--) {
                char c2 = str.charAt(j);
                if(c1 == c2) {
                    String subStr = str.substring(i, j + 1);
                    if(subStr.length() > 2 && isPalindrome(subStr)) {
                        palindromicSubstringCounter.put(subStr.length(), subStr);
                    }
                }
            }
        }
        if(palindromicSubstringCounter.isEmpty()) {
            return "none";
        }
        return palindromicSubstringCounter.get(palindromicSubstringCounter.lastKey());
    }

}

File: Strings/Palindromic_Substring/palindromicSubstring.js

function palindromicSubstring(str) {
    function isPalindrome(str) {
        for(let i = 0, j = str.length - 1; i <= j; i++, j--) {
            if(str.charAt(i) != str.charAt(j)) {
                return false;
            }
        }
        return true;
    }    
    let palindromicSubstr = "";
    let maxPalindromicSubstrLength = 0;
    for(let i = 0; i < str.length; i++) {
        const char1 = str.charAt(i);
        for(let j = str.length - 1; j >= i; j--) {
            const char2 = str.charAt(j);
            if(char1 === char2) {
                const subStr = str.substring(i, j + 1);
                if(isPalindrome(subStr)) {
                    if(subStr.length > maxPalindromicSubstrLength) {
                        maxPalindromicSubstrLength = subStr.length;
                        palindromicSubstr = subStr;
                    }
                }
            }
        }
    }
    if(!palindromicSubstr) {
        return "none";
    }
    return palindromicSubstr;
}

console.log(palindromicSubstring("hellosannasmith") === "sannas");
console.log(palindromicSubstring("abcdefgg") === "none");